test/common/crudtester.go

```diff
@@ -482,8 +483,8 @@ func (c *FederatedTypeCrudTester) waitForResource(client util.ResourceClient, qu
 					}
 					// Lacking type information for the override
 					// field, use string conversion as a cheap way to
-					// determine equality.
-					if fmt.Sprintf("%v", expectedValue) != fmt.Sprintf("%v", value) {
+					// determine equality.  Fall back on reflection.
+					if fmt.Sprintf("%v", expectedValue) != fmt.Sprintf("%v", value) && !reflect.DeepEqual(expectedValue, value) {
```

Comment:
I've updated the comment to provide more detail about why string conversion is attempted.  The tl;dr is that the use of interface types and different deserialization paths breaks comparison by reflection for numeric types, and that comparison of the results of string conversion work around that problem.  Note that this is a test-specific issue, since the sync controller doesn't know enough about any given resource to perform comparison on more than the generation and resource version fields.